include/Model/Languages.php

```diff
@@ -734,6 +734,33 @@ public function clean_cache(): void {
 		$this->cache->clean();
 	}
 
+	/**
+	 * Deletes the transient from the options table since WordPress does not do it when using object cache.
+	 *
+	 * @since 3.8
+	 *
+	 * @return void
+	 */
+	public function delete_transient_from_options_table(): void {
+		if ( wp_using_ext_object_cache() || wp_installing() ) {
+			delete_option( '_transient_' . self::TRANSIENT_NAME );
+		}
+	}
+
+	/**
+	 * Sets the transient in the options table since WordPress does not do it when using object cache.
+	 *
+	 * @since 3.8
+	 *
+	 * @param mixed $value Value of the transient.
+	 * @return void
+	 */
+	public function set_transient_to_options_table( $value ): void {
+		if ( wp_using_ext_object_cache() || wp_installing() ) {
+			add_option( '_transient_' . self::TRANSIENT_NAME, $value, '', true );
+		}
+	}
```

Comment:
Is it useful? I we previously deleted the transient and at some point the external is not used, [the option will be automatically re-created](https://github.com/polylang/polylang/blob/3.7-rc1/include/Model/Languages.php#L536-L538). Am I wrong?

Reply:
Ass discussed during our meeting, this is safer to avoid race conditions between cache and DB.

Reply:
We eventually chose to relay only on transient deletion to keep the DB up to date.

So I removed `set_transient_to_options_table()` in 9d2816186f2e66c18bc9186c7383fcf5a061cd4a.

I also ensure the test fails correctly with this change (if the fix is removed), see ad1608524f5f8df74bd1bcbf99b6d302db10dc86.

Then I took the opportunity to create a dedicated testcase in 9e461881e869c6a5245ce1cdc20b4f3f9698b0cb, indeed I'm currently working on the same issue in Polylang Pro and I could use a testcase for it.